release_tools/changelog.py

```diff
@@ -211,13 +212,42 @@ def create_changelog_entry_content(title, category, author=None, issue=None,
     return content
 
 
-def write_changelog_entry(dirpath, title, content, overwrite=False):
-    """Store the contents of an entry in a file."""
+def validate_changelog_content(content):
+    """Validate the contents of an entry in a file."""
 
     if not content:
         msg = "Aborting due to empty entry content"
         raise click.ClickException(msg)
 
+    try:
+        data = yaml.safe_load(content)
+        return True
+    except yaml.YAMLError as exc:
+        pm = exc.problem_mark
+        click.echo("Error: Invalid format on line {} at position {}".format(pm.line, pm.column))
+        return False
+
+
+def validate_changelog_entry(content):
+    is_valid = False
+
+    while not is_valid:
+        is_valid = validate_changelog_content(content)
+
+        if not is_valid:
+            if not click.confirm("Do you want to edit it?"):
+                msg = "Aborting due to wrong format"
+                raise click.ClickException(msg)
```

Comment:
`confirm` can raise an `Abort` exception when `abort` parameter is set. Wouldn't it be better?

Reply:
Yeah, makes sense.